src/pages/CoinListPage/CoinListPage.tsx

```diff
@@ -1,114 +1,100 @@
-import { useState, useEffect } from "react";
+import { useCallback, useEffect } from "react";
 
 import Dropdown from "@components/UI/Dropdown";
 import Loader from "@components/UI/Loader";
 import Search from "@components/UI/Search";
-import useDropdown from "@hooks/useDropdown";
-import useSearchInput from "@hooks/useSearchInput";
-import { Coin } from "@type/CoinType";
-import axios from "axios";
-import classNames from "classnames";
+import coinListStore from "@store/coinListStore";
+import { Meta } from "@utils/meta";
+import { useAsync } from "@utils/useAsync";
+import { useLocalStore } from "@utils/useLocalStore";
+import { observer } from "mobx-react-lite";
+import { useNavigate } from "react-router-dom";
 
 import styles from "./CoinListPage.module.scss";
 import CoinList from "./components/CoinList";
 
 const CoinListPage = () => {
-  const { search, handleInput, handlerSearchButton } = useSearchInput();
+  const store = useLocalStore(() => new coinListStore());
+  useAsync(store.fetch, [store.dropdownStore.dropdownValues]);
 
-  const [coins, setCoins] = useState<Coin[]>([]);
-
-  const [coinInfoIsLoading, setCoinInfoIsLoading] = useState(false);
-
-  const [currencyInfoIsLoading, setCurrencyInfoIsLoading] = useState(false);
+  const handlerDropdown = useCallback(
+    (args: any) => {
+      store.dropdownStore.dropdownValues = args;
+    },
+    [store.dropdownStore]
```

Comment:
не надо целый стор пихать в зависимости `useCallback`
```suggestion
    [store.dropdownStore.dropdownValues]
```
аналогично для `handleInput` и `handleButton`